lib/active_record_extended/query_methods/either.rb

```diff
@@ -60,4 +60,6 @@ def xor_field_options_for_associations(associations)
   end
 end
 
-ActiveRecord::Base.extend(ActiveRecordExtended::QueryMethods::Either)
+ActiveSupport.on_load :active_record do
+  extend(ActiveRecordExtended::QueryMethods::Either)
```

Comment:
```suggestion
  ActiveRecord::Base.extend(ActiveRecordExtended::QueryMethods::Either)
```

We need to ensure its targeting the correct class when performing the extention.

Reply:
I believe the scope of that block is already `ActiveRecord::Base`.

In the API docs they use `self` to do assignment without repeating the `ActiveRecord::Base` class:
https://api.rubyonrails.org/classes/ActiveSupport/LazyLoadHooks.html

Here's how ActionText hooks into ActiveRecord:
https://github.com/rails/rails/blob/0c2a94c265b22357daecfbaff25501a5b64cea56/actiontext/lib/action_text/engine.rb#L23-L26

I'm fine adding this suggestion (or `self.extend`), but I'm not sure if it functionally makes a difference.

Separately, I'm not sure what's hanging up the tests to help give us confirmation at to whether there's an issue here. Perhaps it needs to be configured to run on pull requests too?
https://github.com/GeorgeKaraszi/ActiveRecordExtended/blob/4222f88453f315cad8fa58b8866326ac7fd2e4e6/.github/workflows/test.yml#L2

I think it's as simple as changing that to `on: [push, pull_request]`

Reply:
Alternatively, what intrigues me is why is this the only case where it's extended directly onto `ActiveRecord::Base`? Could/should it prepend onto `ActiveRecord::Relation`, like `WithCTE`:
https://github.com/GeorgeKaraszi/ActiveRecordExtended/blob/4222f88453f315cad8fa58b8866326ac7fd2e4e6/lib/active_record_extended/query_methods/with_cte.rb#L150

Looking at all the query method classes, they're either prepended to `ActiveRecord::Relation` or `ActiveRecord::QueryMethods::WhereChain`, then there's this one that's extended onto `ActiveRecord::Base`. Is that intentional? Perhaps this was the "old" way of doing it (`Either` seems to be the oldest query method in this gem), and additions since then user a newer approach of prepending to `ActiveRecord::Relation`?

Reply:
Gotcha.

There was an earlier comment I left about your suggested changes that you may have missed.

Reply:
ugh sorry for glancing over some of your comments! Seems you might be correct 👍